### PR TITLE
game: Add arena state hash

### PR DESCRIPTION
**Pull request by mrannanj**

### CODE REVIEW

File: src/utils/log.c [hunk 1]
```diff
@@ -32,15 +33,16 @@ void log_hide(char mode, const char *fn, const char *fmt, ...) {
...
-        fprintf(handle, "[%7u][%c] ", _log_tick, mode);
+        len = snprintf(log_buf, sizeof(log_buf), "[%7u][%c] ", _log_tick, mode);
     }
     va_list args;
     va_start(args, fmt);
-    vfprintf(handle, fmt, args);
+    vsnprintf(&log_buf[len], sizeof(log_buf) - len, fmt, args);
     va_end(args);
-    fprintf(handle, "\n");
+    fprintf(handle, "%s\n", log_buf);
```
Comment (turol) [inline src/utils/log.c:46]: Why not `fputs`?

Comment (mrannanj) [reply src/utils/log.c:46]: Well, the previous code was using this stream handle thing .. dunno

Comment (mrannanj) [reply src/utils/log.c:46]: In theory, the log could be going to a file that is not stdout.

Comment (mrannanj) [reply src/utils/log.c:46]: Oh right, fputs does have `FILE*` argument .. not sure it's worth changing ..

Comment (mrannanj) [reply src/utils/log.c:46]: Okay, so how to add the ending newline nicely if we are using fputs?

Comment (turol) [reply src/utils/log.c:46]: `fputc`

Comment (mrannanj) [reply src/utils/log.c:46]: Nah, would make for 2 writes.

Comment (turol) [reply src/utils/log.c:46]: That should not matter, `FILE *` functions are buffered.

**Pull request merged**